dotnet/src/Connectors/Connectors.AzureOpenAI/Core/AzureClientCore.ChatCompletion.cs

```diff
@@ -31,19 +33,36 @@ protected override OpenAIPromptExecutionSettings GetSpecializedExecutionSettings
     protected override Dictionary<string, object?> GetChatCompletionMetadata(OpenAIChatCompletion completions)
     {
 #pragma warning disable AOAI001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
-        return new Dictionary<string, object?>
+        var metadata = new Dictionary<string, object?>
         {
             { nameof(completions.Id), completions.Id },
             { nameof(completions.CreatedAt), completions.CreatedAt },
-            { ContentFilterResultForPromptKey, completions.GetContentFilterResultForPrompt() },
             { nameof(completions.SystemFingerprint), completions.SystemFingerprint },
             { nameof(completions.Usage), completions.Usage },
-            { ContentFilterResultForResponseKey, completions.GetContentFilterResultForResponse() },
-
             // Serialization of this struct behaves as an empty object {}, need to cast to string to avoid it.
             { nameof(completions.FinishReason), completions.FinishReason.ToString() },
             { nameof(completions.ContentTokenLogProbabilities), completions.ContentTokenLogProbabilities },
         };
+
+        try
+        {
+            metadata[ContentFilterResultForPromptKey] = completions.GetContentFilterResultForPrompt();
+            metadata[ContentFilterResultForResponseKey] = completions.GetContentFilterResultForResponse();
+        }
+        catch (MissingFieldException ex)
```

Comment:
Wasn't the issue fixed in the SDK? If it was not, please add a short comment pointing to the GitHub issue so we can later check its status and remove the catch block when the issue is resolved.

Reply:
This error only may happen if the version used in `Azure.AI.OpenAI` is older. I may be able to remove those if we garantee this package will never run against an older than beta.3 Azure.Ai.OpenAI SDK.